typescript/ccip-server/src/services/CallCommitmentsService.ts

```diff
@@ -0,0 +1,301 @@
+      salt,
+      ica,
+      revealMessageId,
+      commitmentDispatchTx,
+      originDomain,
+    } = data;
+    await prisma.commitment.create({
+      data: {
+        commitment,
+        revealMessageId,
+        calls,
+        relayers,
+        salt,
+        ica,
+        commitmentDispatchTx,
+        originDomain,
+      },
+    });
+    console.log('Stored commitment', { commitment, ...data });
+  }
+
+  /**
+   * Fetch a commitment record from the database by ID.
+   * Throws if not found.
+   */
+  private async fetchCommitmentRecord(
+    commitment: string,
+    revealMessageId: string,
+  ) {
+    console.log('Fetching commitment from DB', commitment, revealMessageId);
+    const record = await prisma.commitment.findUnique({
+      where: { commitment_revealMessageId: { commitment, revealMessageId } },
+    });
+    if (!record) {
+      console.log('Commitment not found in DB', commitment, revealMessageId);
+      throw new Error('Commitment not found');
+    }
+    return record;
+  }
+
+  // Validate the commitment events by checking the transaction receipt
+  // and parsing the events emitted by the InterchainAccountRouter.
+  // This ensures that the commitment was dispatched correctly and
+  // return the ICA address.
+  // Throws if validation fails.
+  private async validateCommitmentEvents(
+    data: PostCallsType,
+    commitment: string,
+  ): Promise<{ ica: string; revealMessageId: string }> {
+    const provider = this.multiProvider.getProvider(data.originDomain);
+    const receipt = await provider.getTransactionReceipt(
+      data.commitmentDispatchTx,
+    );
+    if (!receipt)
+      throw new Error(
+        `Transaction not found: ${data.commitmentDispatchTx} on domain ${data.originDomain}`,
+      );
+
+    // 2) Verify reveal event
+    this.ensureCommitmentDispatched(receipt, commitment);
+
+    // 3) Extract reveal message ID
+    const revealMessageId = this.extractRevealMessageId(receipt);
+
+    // 4) Derive ICA from RemoteCallDispatched
+    const ica = await this.deriveIcaFromRemoteCallDispatched(
+      receipt,
+      data.originDomain,
+    );
+    return { ica, revealMessageId };
+  }
+
+  /**
+   * Ensure a CommitRevealDispatched event matching the commitment exists.
+   */
+  private ensureCommitmentDispatched(receipt: any, commitment: string): void {
+    const iface = InterchainAccountRouter__factory.createInterface();
+    const revealTopic = iface.getEventTopic('CommitRevealDispatched');
+    const revealLogs = receipt.logs.filter(
+      (l: any) => l.topics[0] === revealTopic,
+    );
+    if (revealLogs.length === 0) {
+      throw new Error('CommitRevealDispatched event not found');
+    }
+
+    const matched = revealLogs
+      .map((l: any) => iface.parseLog(l))
+      .some((parsed: any) => parsed.args.commitment === commitment);
+
+    if (!matched) {
+      const foundCommitments = revealLogs.map(
+        (l: any) => iface.parseLog(l).args.commitment,
+      );
```

Comment:
dont use any types here?
you are also mapping over the logs twice

Reply:
I believe we discussed this and typechain doesn't actually expose these types without using filters for the rpc

f50eea4cd